.github/workflows/kind-e2e.yaml

```
@@ -0,0 +1,65 @@
+name: kind-e2e
+on:
+  # enable running this on merges + presubmits when the action has tests that it will execute
+  # push:
+  #   branches: [main]
+  # pull_request:
+  workflow_dispatch:
+jobs:
+  test-kind:
+    permissions:
+      issues: write
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        k8sVersion: ["1.23.x", "1.24.x", "1.25.x", "1.26.x", "1.27.x", "1.28.x", "1.29.x", "1.30.x"]
+    steps:
+    - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+    - uses: ./.github/actions/install-deps
+      with:
+          k8sVersion: ${{ matrix.k8sVersion }}
+    - name: Kind Cluster
+      uses: helm/kind-action@0025e74a8c7512023d06dc019c617aa3cf561fde # v1.10.0
+    - name: check kind cluster and taint nodes
+      shell: bash
+      run: |
+        kubectl config current-context
+        kubectl get nodes
+        kubectl taint nodes chart-testing-control-plane CriticalAddonsOnly:NoSchedule
+    - name: Enable the actionlint matcher
+      shell: bash
+      run: echo "::add-matcher::.github/actionlint-matcher.json"
+    - name: install pyroscope
+      shell: bash
+      run: | 
+        curl -fL https://github.com/grafana/pyroscope/releases/download/v1.1.5/profilecli_1.1.5_linux_amd64.tar.gz | tar xz profilecli
```

```suggestion
        curl -fL https://github.com/grafana/pyroscope/releases/download/v1.1.5/profilecli_1.1.5_linux_amd64.tar.gz | tar xz pyroscope
```

nit: Why not call it pyroscope?